### PR TITLE
[Product blocks editor]: Add help text to Shipping toggle

### LINKED ISSUE #41509: Add help text to Shipping toggle
We want to provide users with more information so they know what to expect when they toggle off shipping/pickup.
Let's add help text when `This product requires shipping or pickup` is **toggled off.**

**Help Text to be added:**

`This product will not trigger your customer’s shipping calculator in cart or at checkout. This product also won’t require your customers to enter their shipping details at checkout. [Read more about virtual products.](https://woo.com/document/managing-products/#adding-a-virtual-product)`

**Current UI:**
image

**Proposed UI:**
image

Note that we don't want this help text to appear when the toggle is on. So when toggled on, it should still look like this: 
image

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

The changes introduced by this PR show a help section in the shipping section of the Product/Variation.

Closes https://github.com/woocommerce/woocommerce/issues/41509

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the product editor
2. Ensure to create a product/variations 
3. Go to the Shipping section

Before | After
-------|------
image | image

https://github.com/woocommerce/woocommerce/assets/77539/83b30963-50da-4cbd-9061-44811d847093

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/blocks/generic/toggle/edit.tsx [hunk 1]
```diff
@@ -64,13 +64,26 @@ export function Edit( {
...
+	 * Create a Help component in case
+	 * the help attribute is set.
+	 */
+	const help = attributes?.help
+		? sanitizeHTML( attributes.help )?.__html
+		: null;
+
+	const helpComponent = createElement( 'div', {
+		dangerouslySetInnerHTML: { __html: help },
+	} );
```
Comment (louwie17) [inline packages/js/product-editor/src/blocks/generic/toggle/edit.tsx]: Nice!

File: packages/js/product-editor/src/blocks/generic/toggle/block.json [hunk 2]
```diff
@@ -12,6 +12,9 @@
 			"type": "string",
 			"__experimentalRole": "content"
 		},
+		"help": {
+			"type": "string"
+		},
```
Comment (louwie17) [inline packages/js/product-editor/src/blocks/generic/toggle/block.json:17]: One of the criteria's was to hide the `help` text when toggled on:

> Note that we don't want this help text to appear when the toggle is on. So when toggled on, it should still look like this:

Should we provide help text for when it's toggled on or off?

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks for working on this @retrofox the changes look good! 
This change does always show the help text, and it looks like we should not show the help text when it is enabled:
> Note that we don't want this help text to appear when the toggle is on. So when toggled on, it should still look like this:

Are you able to add that logic in? Maybe the best way is to have an `help` parameter for when it's toggled on/off?

File: packages/js/product-editor/src/blocks/generic/toggle/block.json (hunk 2 above)
Comment (retrofox) [reply packages/js/product-editor/src/blocks/generic/toggle/block.json:17]: if we want to make it scalable, probably we'll have to add another attribute, like `helpOff` or something like that

**Comment by retrofox** [comment]
> This change does always show the help text, and it looks like we should not show the help text when it is enabled:
> > Note that we don't want this help text to appear when the toggle is on. So when toggled on, it should still look like this:

👍 Addressed here
 
> Are you able to add that logic in? Maybe the best way is to have an `help` parameter for when it's toggled on/off?

Yes, we could do that in case it gets a requirement?

File: packages/js/product-editor/src/blocks/generic/toggle/block.json (hunk 2 above)
Comment (louwie17) [reply packages/js/product-editor/src/blocks/generic/toggle/block.json:17]: Yeah I did prefer the more scalable option.
Maybe we could follow the format of the `checkedValue`/`uncheckedValue` and use `checkedHelpText` & `uncheckedHelpText`? or just `checkedHelp` would work to, the `text` addition might be slightly more clear.

File: packages/js/product-editor/src/blocks/generic/toggle/edit.tsx [hunk 3]
```diff
@@ -64,13 +64,26 @@ export function Edit( {
...
+
 	return (
 		<div { ...blockProps }>
 			<ToggleControl
 				label={ label }
 				checked={ isChecked() }
 				disabled={ disabled }
 				onChange={ handleChange }
+				help={ isChecked() ? null : helpComponent }
```
Comment (louwie17) [inline packages/js/product-editor/src/blocks/generic/toggle/edit.tsx]: I think we should go with the more scalable approach, as this logic is very specific for a generic `toggle` block/field.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks for the update @retrofox, this tested well!
I do think we should go with the scalable solution as you suggested, especially as this is a generic field/block.

**Comment by retrofox** [comment]
> Thanks for the update @retrofox, this tested well! I do think we should go with the scalable solution as you suggested, especially as this is a generic field/block.

Yes, good idea. Already addressed :-)

**Review by louwie17 — APPROVED** [review]
Nice work @retrofox, this tested well and code looks good, thanks for the updates 👍 💯

**Pull request merged**